nameable.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Implement!!!'
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = @nameable.correct_name.slice(0, 10)
+    name
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** According to ruby conventions, classes should be in their own files. I would recommend you move these classes to their respective files

Reply:
Thanks will change it